netbsd/libs.md

```diff
@@ -0,0 +1,143 @@
+# Instructions for obtaining EFI related libraries on NetBSD
+
+Read [README.md](./README.md) first for general information.
+
+NetBSD already has support for EFI variables in the kernel, but this
+functionality is not yet available in releases, which is why we need to build
+the kernel ourselves.
+
+The plan is:
+ * install kernel with support for EFI variables
+ * prepare environment for building ports
+ * build and install `efivar` port
+ * build a test tool
+ * verify that it works
+
+## Install kernel with support for EFI variables
+
+You can either cross-compile and make an ISO as described in the README or run
+the commands below as `root` on an already running NetBSD:
+
+```
+# install git
+export PATH="/usr/pkg/sbin:/usr/pkg/bin:$PATH"
+export PKG_PATH="http://cdn.netbsd.org/pub/pkgsrc/packages/NetBSD/amd64/9.3/All/"
+pkg_add git
+
+# clone sources
+export GIT_SSL_NO_VERIFY=true
+git clone -b os-interface --depth 1 https://github.com/3mdeb/NetBSD-src.git /usr/src
+
+# build the kernel
+cd /usr/src/sys/arch/amd64/conf
+cp GENERIC MYKERNEL
+mkdir /usr/obj
+cd /usr/src
+./build.sh tools
+./build.sh -u -j$(sysctl -n hw.ncpu) kernel=MYKERNEL
+
+# install as a second kernel to be used when needed
+cp /usr/obj/sys/arch/amd64/compile/MYKERNEL/netbsd /netbsd.my
+
+# create /dev/efi device file
+make -C /usr/src/etc MAKEDEV TOOLDIR=/usr/obj/tooldir.NetBSD-9.3-amd64/ MKDTB=no
+cd /dev
+sh /usr/src/etc/MAKEDEV efi
+
+# restart the system to load a new kernel
+reboot
+# press 3 and then run "boot netbsd.my" on > prompt
+```
+
+## Preparing for using ports
+
+See above how to install `git`. Run this and other commands below as a `root`
+user:
+
+```
+git clone -b efivar --depth=1 https://github.com/3mdeb/pkgsrc /usr/pkgsrc
+```
+
+## Building and installing the port
+
+```
+cd /usr/pkgsrc/devel/efivar/
+# update "distinfo" file in case it diverged due to active development
+make makesum
+# build and install the port
+make install
```

Comment:
I get this error while compiling:

```
ioctl.c:25:10: fatal error: sys/efiio.h: No such file or directory
```

Reply:
Added `cp /usr/src/sys/sys/efiio.h /usr/include/sys/` which I thought shouldn't be needed if using non-official ISO image.